src/EditorFeatures/CSharpTest/ConvertToRecord/ConvertToRecordTests.cs

```diff
@@ -928,17 +983,1279 @@ public class [|C|]
+}
+";
+            var changedMarkup = @"
+namespace N
+{
+    public record C(int P, bool B);
+}
+";
+            await TestRefactoringAsync(initialMarkup, changedMarkup).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestMovePropertiesAndDeleteAsCastEqualsWithIsNotNull()
+        {
+            var initialMarkup = @"
+namespace N
+{
+    public class [|C|]
+    {
+        public int P { get; init; }
+        public bool B { get; init; }
+
+        public override bool Equals(object? other)
+        {
+            var otherC = other as C;
+            return otherC is not null && otherC.P == P && otherC.B == B;
+        }
+    }
+}
+";
+            var changedMarkup = @"
+namespace N
+{
+    public record C(int P, bool B);
+}
+";
+            await TestRefactoringAsync(initialMarkup, changedMarkup).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestMovePropertiesAndKeepAsCastEqualsWithIncorrectIsNull()
+        {
+            var initialMarkup = @"
+namespace N
+{
+    public class [|C|]
+    {
+        public int P { get; init; }
+        public bool B { get; init; }
+
+        public override bool Equals(object? other)
+        {
+            var otherC = other as C;
+            return otherC is null && {|CS8602:otherC|}.P == P && otherC.B == B;
+        }
+    }
+}
+";
+            var changedMarkup = @"
+namespace N
+{
+    public record C(int P, bool B)
+    {
+
+        public override bool {|CS0111:Equals|}(object? other)
+        {
+            var otherC = other as C;
+            return otherC is null && {|CS8602:otherC|}.P == P && otherC.B == B;
+        }
+    }
+}
+";
+            await TestRefactoringAsync(initialMarkup, changedMarkup).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestMovePropertiesAndDeleteAsCastEqualsWithIsNull()
+        {
+            var initialMarkup = @"
+namespace N
+{
+    public class [|C|]
+    {
+        public int P { get; init; }
+        public bool B { get; init; }
+
+        public override bool Equals(object? other)
+        {
+            var otherC = other as C;
+            return !(otherC is null || otherC.P != P || otherC.B != B);
+        }
+    }
+}
+";
+            var changedMarkup = @"
+namespace N
+{
+    public record C(int P, bool B);
+}
+";
+            await TestRefactoringAsync(initialMarkup, changedMarkup).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestMovePropertiesAndDeleteSimpleTypeEquals()
+        {
+            var initialMarkup = @"
+using System;
+
+namespace N
+{
+    public class [|C|] : IEquatable<C>
+    {
+        public int P { get; init; }
+        public bool B { get; init; }
+
+        public bool Equals(C? otherC)
+        {
+            return {|CS8602:otherC|}.P == P && otherC.B == B;
+        }
+    }
+}
+";
+            var changedMarkup = @"
+using System;
+
+namespace N
+{
+    public record C(int P, bool B) : IEquatable<C>;
```

Comment:
Should we be removing the `IEquatable<C>` in this case? We're getting that for free at this point.